kyo-data/shared/src/main/scala/kyo/Tag.scala

```diff
@@ -1,239 +1,541 @@
+          * @return
+          *   true if this type is a subtype of the other type, false otherwise
+          */
+        infix def <:<[B](that: Tag[B]): Boolean =
+            (self eq that) || self.equals(that) || isSubtype(self, that)
+
+        /** Tests if this Tag represents a supertype of another Tag. This is the reverse of the <:< operation.
+          *
+          * @param that
+          *   The Tag to compare with
+          * @return
+          *   true if this type is a supertype of the other type, false otherwise
+          */
+        infix def >:>[B](that: Tag[B]): Boolean =
+            that <:< self
+
+        /** Creates a Tag representing the intersection of this type and another type. The resulting Tag represents values that conform to
+          * both type A and type B.
+          *
+          * @param that
+          *   The Tag to intersect with
+          * @return
+          *   A Tag representing the intersection type A & B
+          */
+        infix def &[B](that: Tag[B]): Tag[A & B] =
+            Type.IntersectionType(self.tpe, that.tpe)
+
+        /** Creates a Tag representing the union of this type and another type. The resulting Tag represents values that conform to either
+          * type A or type B.
+          *
+          * @param that
+          *   The Tag to union with
+          * @return
+          *   A Tag representing the union type A | B
+          */
+        infix def |[B](that: Tag[B]): Tag[A | B] =
+            Type.UnionType(self.tpe, that.tpe)
+
+        /** Erases the specific type information, returning a Tag for Any. This operation is useful when you need to work with Tags of
+          * different types uniformly.
+          *
+          * @return
+          *   A Tag[Any] representing the same type structure but with type parameter erased
+          */
+        def erased: Tag[Any] = self.asInstanceOf[Tag[Any]]
+
+        /** Computes a hash code for this Tag based on its type structure. This hash is used in the caching system for subtype checking.
+          *
+          * @return
+          *   A hash code for this Tag
+          */
+        def hash: Int =
+            self.tpe.hashCode()
+
+        /** Retrieves the decoded Type representation of this Tag. If the Tag is already a Type, it is returned directly. If it's an encoded
+          * string, it is decoded (with caching) and then returned.
+          *
+          * @return
+          *   The Type representation of this Tag
+          */
+        def tpe: Type[A] =
+            self match
+                case self: String  => decode(self)
+                case self: Type[A] => self
+
+        /** Returns a human-readable representation of the type structure. This is useful for debugging and understanding complex type
+          * relationships.
+          *
+          * @return
+          *   A string representation of the type
+          */
+        def show: String =
+            self.tpe.toString()
```

Comment:
I've removed `showTpe` and made `show` print the type directly instead. The distinction wasn't very useful.